lib/Yield/CrossRelations.cpp

```diff
@@ -0,0 +1,168 @@
+/// \file CallGraph.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <unordered_map>
+
+#include "revng/ADT/STLExtras.h"
+#include "revng/Model/Binary.h"
+#include "revng/Pipes/Ranks.h"
+#include "revng/Yield/CrossRelations.h"
+
+using CR = yield::CrossRelations;
+CR::CrossRelations(const SortedVector<efa::FunctionMetadata> &Metadata,
+                   const model::Binary &Binary) {
+  revng_assert(Metadata.size() == Binary.Functions.size());
+
+  namespace rank = revng::pipes::rank;
+
+  for (auto Inserter = Relations.batch_insert();
+       const auto &Function : Binary.Functions) {
+    namespace rank = revng::pipes::rank;
+    const auto Location = rank::Function.makeLocation(Function.Entry);
+    Inserter.insert(yield::RelationDescription(Location.toString(), {}));
+  }
+
+  for (const auto &[EntryAddress, ControlFlowGraph] : Metadata) {
+    auto CallLocation = rank::Instruction.makeLocation(EntryAddress,
+                                                       MetaAddress::invalid(),
+                                                       MetaAddress::invalid());
+
+    for (const auto &BasicBlock : ControlFlowGraph) {
+      for (const auto &Edge : BasicBlock.Successors) {
+        if (efa::FunctionEdgeType::isCall(Edge->Type)) {
+          if (const auto &Callee = Edge->Destination; Callee.isValid()) {
+            // TODO: embed information about the call instruction into
+            //       `CallLocation` after efa starts providing it.
+
+            auto Location = rank::Function.makeLocation(Callee).toString();
+            if (auto It = Relations.find(Location); It != Relations.end()) {
+              yield::RelationTarget T(yield::RelationType::IsCalledFrom,
+                                      CallLocation.toString());
+              It->Related.insert(std::move(T));
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+template<typename AddNodeCallable, typename AddEdgeCallable>
+static void convertionHelper(const yield::CrossRelations &Input,
```

Comment:
`s/convertion/conversion/`

Reply:
https://github.com/revng/revng/pull/197/commits/337ff34e5d97fb0394f7b7c615f9178865b5add9

and also 

https://github.com/revng/revng/pull/197/commits/9f16bed463452f897f87e7d4ad268fefefec8b18